src/test/java/org/prebid/server/auction/ExchangeServiceTest.java

```diff
@@ -1199,27 +1202,31 @@ public void shouldUpdateBidPriceWithCurrencyConversionAndPriceAdjustmentFactor()
 
         given(currencyService.convertCurrency(any(), any(), any(), any())).willReturn(BigDecimal.valueOf(10.0));
 
-        final BigDecimal updatedPrice = BigDecimal.valueOf(100);
-        givenBidResponseCreator(singletonList(Bid.builder().price(updatedPrice).build()));
-
         // when
-        final BidResponse bidResponse = exchangeService.holdAuction(givenRequestContext(bidRequest)).result();
+        exchangeService.holdAuction(givenRequestContext(bidRequest)).result();
 
         // then
-        assertThat(bidResponse.getSeatbid())
-                .flatExtracting(SeatBid::getBid)
-                .extracting(Bid::getPrice)
-                .containsExactly(updatedPrice);
+        @SuppressWarnings("unchecked") final ArgumentCaptor<List<BidderResponse>> argumentCaptor =
+                ArgumentCaptor.forClass(List.class);
+        verify(bidResponseCreator).create(argumentCaptor.capture(), any(), any(), any(), any(), any(), anyBoolean());
+
+        assertThat(argumentCaptor.getValue()).hasSize(1);
+
+        final BigDecimal updatedPrice = BigDecimal.valueOf(100);
+        final Bid expectedBid = Bid.builder().price(updatedPrice).build();
+        final BidderSeatBid firstSeatBid = argumentCaptor.getValue().get(0).getSeatBid();
+        assertThat(firstSeatBid.getBids()).extracting(BidderBid::getBid).containsOnly(expectedBid);
```

Comment:
No point to create an `expectedBid` object if you wan't to compare just a price. You can just do how it was before - .extracting(Bid::getPrice), and then .containsOnly(). Check for the similar cases.